bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/ItcTable.scala

```diff
@@ -243,6 +248,30 @@ trait ItcTable extends Table {
       }
     }
   }
+
+  // Makes an educated guesstimate of the fraction of images of the given set that are on source.
+  // For imaging the field of view is the science area, for spectroscopy the slit width of the mask is used as width.
+  // Then, for every image we check if the p or q offsets are more than 0.5 times the horizontal or vertical
+  // field of view, if so, we assume that image is off-source, otherwise we assume it is on-source.
+  private def extractSourceFraction(uc: ItcUniqueConfig, instrument: SPInstObsComp): String \/ Double = {
+
+    // get science area / field of view in arcsecs of the instrument
+    // this already takes care to replace the width with the slit width of the mask for spectroscopy
+    val scienceArea = instrument.getScienceArea
+    val (w, h)      = (scienceArea(0), scienceArea(1))
```

Comment:
The `getScienceArea` method should probably be deprecated.  It assumes that the science area is always rectangular which clearly isn't the case.  Until we get all the precise science areas into the model though, I guess there's nothing else to go on.